pkg/operator/ceph/cluster/rbd/controller.go

```diff
@@ -284,29 +313,42 @@ func (r *ReconcileCephRBDMirror) reconcileCreateCephRBDMirror(cephRBDMirror *cep
 }
 
 // updateStatus updates an object with a given status
-func (r *ReconcileCephRBDMirror) updateStatus(observedGeneration int64, name types.NamespacedName, status string) {
+func (r *ReconcileCephRBDMirror) updateStatus(observedGeneration int64, name types.NamespacedName, status string, cephx *cephv1.CephxStatus) error {
 	rbdMirror := &cephv1.CephRBDMirror{}
-	err := r.client.Get(r.opManagerContext, name, rbdMirror)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debug("CephRBDMirror resource not found. Ignoring since object must be deleted.")
-			return
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		err := r.client.Get(r.opManagerContext, name, rbdMirror)
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Debug("CephRBDMirror resource not found. Ignoring since object must be deleted.")
+				return nil
+			}
+			return errors.Wrapf(err, "failed to retrieve CephRBDMirror %q to update status to %q", name, status)
 		}
-		logger.Warningf("failed to retrieve rbd mirror %q to update status to %q. %v", name, status, err)
-		return
-	}
 
-	if rbdMirror.Status == nil {
-		rbdMirror.Status = &cephv1.Status{}
-	}
+		if rbdMirror.Status == nil {
+			rbdMirror.Status = &cephv1.RBDMirrorStatus{}
+		}
 
-	rbdMirror.Status.Phase = status
-	if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
-		rbdMirror.Status.ObservedGeneration = observedGeneration
-	}
-	if err := reporting.UpdateStatus(r.client, rbdMirror); err != nil {
-		logger.Errorf("failed to set rbd mirror %q status to %q. %v", rbdMirror.Name, status, err)
-		return
+		rbdMirror.Status.Phase = status
+		if observedGeneration != k8sutil.ObservedGenerationNotAvailable {
+			rbdMirror.Status.ObservedGeneration = observedGeneration
+		}
+
+		if cephx != nil {
+			rbdMirror.Status.Cephx.Daemon = *cephx
+		}
+
+		if err := reporting.UpdateStatus(r.client, rbdMirror); err != nil {
+			return errors.Wrapf(err, "failed to set CephRBDMirror %q status to %+v", name, rbdMirror.Status)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
 	}
-	logger.Debugf("rbd mirror %q status updated to %q", name, status)
+
+	logger.Debugf("successfully updated cephRBDMirror %q status updated to %+v", name, rbdMirror.Status)
```

Comment:
```suggestion
	logger.Debugf("successfully updated cephRBDMirror %q status updated to %+v", namespacedName, rbdMirror.Status)
```

Reply:
`name` variable over here was actually `namespacedName`. Changed `name` in the method signature to `namespacedName` to avoid confusion.